test/AtomicSwap.js

```diff
@@ -0,0 +1,125 @@
+    try {
+      const instance = await atomicSwap.deployed();
+      await instance.initiate(now + day, hashLock, other, {value: 10e18});
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should fail if swap doesn't exist`, async () => {
+    try {
+      const instance = await atomicSwap.deployed();
+      await instance.redeem("some-inexistent-secret");
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it (`should fail if the message sender is not the swap participant`, async () => {
+    try {
+      const instance = await atomicSwap.deployed();
+      await instance.redeem(secret);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it('should let you redeem your ether from the swap', async () => {
+    const instance = await atomicSwap.deployed();
+
+    const prevBalance = await web3.eth.getBalance(other);
+
+    await instance.redeem(secret, {from: other});
+    
+    const balance = await web3.eth.getBalance(other);
+
+    const {4: exists} = await instance.swaps(other, hashLock);
+  
+    assert.equal(balance.sub(prevBalance).valueOf(), 10e18);
+
+    assert.equal(exists, false);
+  });
+
+  it(`should not refund if swap doesn't exist for participant`, async () => {
+    try {
+      const instance = await atomicSwap.deployed();
+      await instance.refund(hashLock, owner);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should not refund if swap doesn't exist for participant with that hash`, async () => {
+    try {
+      const instance = await atomicSwap.deployed();
+      await instance.refund("wrong-hahs-lock", other);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should not refund if exipration time was not met`, async () => {
+    try {
+      const instance = await atomicSwap.deployed();
+      await instance.refund(hashLock, other);
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should not refund if refundee is not the initiator`, async () => {
+    try {
+      const instance = await atomicSwap.deployed();
+
+      await timeTravel(day);
+
+      await instance.refund(hashLock, other, {from: notInitiator});
+    } catch (error) {
+      assert.equal(error.message, 'VM Exception while processing transaction: revert');
+    }
+  });
+
+  it(`should refund if all requirements are met`, async () => {
+    const instance = await atomicSwap.deployed();
+
+    const beforeBalance = await web3.eth.getBalance(owner);
+
+    await timeTravel(day);
+
+    await instance.refund(hashLock, other);
```

Comment:
Throws a revert even tho I don't understand, tested manually on remix and it worked as expected.

Reply:
Have you tried waiting more than a day? Maybe the 'now' parameter in testrpc isn't being updated past the swap creation + 1 day timestamp?

Reply:
I tried 2 days, took the now after timeTravel and the check was fine, somehow in coverage for me the lines for successful refund appears covered.